scene/resources/particles_material.cpp

```diff
@@ -701,24 +807,35 @@ void ParticlesMaterial::_update_shader() {
 			code += "	TRANSFORM[3] = origin;\n";
 		}
 	}
-	//scale by scale
-	code += "	float base_scale = mix(scale_min, scale_max, scale_rand);\n";
-	code += "	base_scale = sign(base_scale) * max(abs(base_scale), 0.001);\n";
 
-	code += "	TRANSFORM[0].xyz *= base_scale * sign(tex_scale.r) * max(abs(tex_scale.r), 0.001);\n";
-	code += "	TRANSFORM[1].xyz *= base_scale * sign(tex_scale.g) * max(abs(tex_scale.g), 0.001);\n";
-	code += "	TRANSFORM[2].xyz *= base_scale * sign(tex_scale.b) * max(abs(tex_scale.b), 0.001);\n";
 	if (particle_flags[PARTICLE_FLAG_DISABLE_Z]) {
 		code += "	VELOCITY.z = 0.0;\n";
 		code += "	TRANSFORM[3].z = 0.0;\n";
 	}
+
 	if (collision_enabled) {
 		code += "	if (COLLIDED) {\n";
-		code += "		TRANSFORM[3].xyz+=COLLISION_NORMAL * COLLISION_DEPTH;\n";
-		code += "		VELOCITY -= COLLISION_NORMAL * dot(COLLISION_NORMAL, VELOCITY) * (1.0 + collision_bounce);\n";
-		code += "		VELOCITY = mix(VELOCITY,vec3(0.0),collision_friction * DELTA * 100.0);\n";
+		code += "		if (length(VELOCITY) > 3.0) {\n";
```

Comment:
Can you please remind me why this change was made?

Reply:
It's a tricky one... it's a fix/workaround for the really far from perfect collision implementation.

Collision is something that I found more than whacky at the moment, strongly dependent on the particle system fps.
With the default 30 fps and some gravity, particles never come to a state where they stay in the "Collided" state and so the Turbulence continues to be applied. That looks really wrong and so this workaround was introduced.

When the collision implementation is getting a bit more love I'm pretty sure we can remove this.

https://user-images.githubusercontent.com/9423774/178154251-f5056aa4-c6e6-4042-ace9-e0c332cf1983.mp4
(Example of particles never coming to rest - no turbulence applied, just defaults. Red = !Collided, Green = Collided)

https://user-images.githubusercontent.com/9423774/178154294-3b901f42-3329-46c9-bddd-8e58f19b692b.mp4
(Example of how it would look without the workaround)

https://user-images.githubusercontent.com/9423774/178154307-02138d72-cc58-43d7-9e99-ef2ba1591f18.mp4
(Example with the workaround)


Overall collision needs some more work before it can considered "ready to use". It's pretty obvious when you start playing around with it.

Reply:
https://github.com/godotengine/godot/pull/61245 should improve this significantly.

Reply:
I agree with the default of 30 being a bad choice. I will check the PR.